### PR TITLE
Fix: #276 - Add ability to set custom User Agent

### PR DESCRIPTION
**Pull request by jhonatn**
It'd be useful to set a custom UA for identifying desktop apps. Additionally, some sites require UA spoofing to work.

### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information
- Based on the [draft PR 280](https://github.com/tauri-apps/wry/pull/280) from @aditsachde 
- The Windows implementation still has to be tested

### CODE REVIEW

File: src/webview/webview2/mod.rs [hunk 1]
```diff
@@ -471,6 +471,13 @@ impl InnerWebView {
       }
     }
 
+    // Set user agent
+    if let Some(user_agent) = attributes.user_agent {
+      unsafe {
+        webview.Settings()?.SetUserAgent(String::from(user_agent.as_str()));
```
Comment (wusyong) [inline src/webview/webview2/mod.rs]: Looks like I don't have the right to push your PR.
Could you add these lines for me?
```suggestion
        let settings: ICoreWebView2Settings2 = webview.Settings()?
          .cast()
          .map_err(webview2_com::Error::WindowsError)?;
        settings.SetUserAgent(String::from(user_agent.as_str()))?;
```

**Review by wusyong — CHANGES_REQUESTED** [review]

File: src/webview/webview2/mod.rs (hunk 1 above)
Comment (wusyong) [reply src/webview/webview2/mod.rs]: and probably run `cargo fmt` after this

**Comment by lucasfernog** [comment]
We need to check if this has security implications, if so we need to push it into v2.

File: src/webview/webview2/mod.rs (hunk 1 above)
Comment (jhonatn) [reply src/webview/webview2/mod.rs]: Done!

**Comment by wusyong** [comment]
Should be fine if tauri don't expose it on v1 I suppose. I'll merge this in wry first.

**Pull request merged**